src/OrchardCore.Modules/OrchardCore.Liquid/Drivers/LiquidPartDisplay.cs

```diff
@@ -38,7 +38,19 @@ public override IDisplayResult Edit(LiquidPart liquidPart)
 
         public override async Task<IDisplayResult> UpdateAsync(LiquidPart model, IUpdateModel updater)
         {
-            await updater.TryUpdateModelAsync(model, Prefix, t => t.Liquid);
+            var viewModel = new LiquidPartViewModel();
+
+            if (await updater.TryUpdateModelAsync(viewModel, Prefix, t => t.Liquid))
+            {
+                if (!string.IsNullOrEmpty(viewModel.Liquid) && !_liquidTemplatemanager.Validate(viewModel.Liquid, out var errors))
+                {
+                    updater.ModelState.AddModelError(nameof(model.Liquid), T["The 'Body' contains an invalid Liquid expression. Details: {0}", string.Join(" ", errors)]);
```

Comment:
We can't, it will be html encoded.

As i remember this is for the same reason that we moved some localizations from tag attributes values to tag contents. But here, we can't move the error message to another place.

Note: Seems that the liquid validation ends on the 1st error, as i have tried there is always only one error.

Reply:
> We can't, it will be html encoded.

Hmm we did something similar in delete feature confirm dialog